### ISSUE TITLE
Move Social Icons size setting to the inspector

### ISSUE THREAD

**Issue by afercia** [WordPress/gutenberg#65647]
### Description

The Social Icons block has a 'size' setting that is placed in a toolbar dropdown menu labeled 'Size':

![Screenshot 2024-09-25 at 12 40 06](https://github.com/user-attachments/assets/e45c91a3-7351-4a61-bcf7-16b0ab9082ac)

![Screenshot 2024-09-25 at 14 02 44](https://github.com/user-attachments/assets/20fee141-241f-4015-8b72-a88b3fc3c1c8)

I may be missing something but I couldn't think of any other 'size' setting placed in the block toolbar. All the 'size' settings I can think of are placed in the settings panel (the Inspector).
The Buttons block is very similar to the Social Icons one, which has its Size setting in the Insepctor.
Other size settings are in the Inspector, whether they're typograhy global sizes or the font size picker for some blocks.

As a user, I learnt to associate the concept of 'size' with the setting panel and I'd expect to find this setting always in the settings panel.

Is there any design reason to place the Social Icons size setting in the block toolbar? To me, this seems an unique case, an inconsistency that doesn't help provide a cohesive, consistent experience to users.

Cc @WordPress/gutenberg-design 

### Step-by-step reproduction instructions

- Add a Social Icons block.
- Observe the Size setting is in the block toolbar.
- Add a Buttons block.
- Observe the Size setting is in the settings panel (the Inspector).
- Observe other 'Size' settings are placed in the settings panel (the Inspector).

### Please confirm that you have searched existing issues in the repo.

- [X] Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

- [X] Yes

**Comment by hanneslsm** [comment]
> I may be missing something but I couldn't think of any other 'size' setting placed in the block toolbar. 

Not size, but the search block has also display options in the toolbar. (Something that should be revisited also, though.)

**Comment by richtabor** [comment]
Curious, why is this labeled for accessibility?

**Comment by richtabor** [comment]
I agree, there's potential in moving this to the inspector (or potentially having both placements—not 100% sure on that though). 

Are you thinking a ToggleGroup control, similar to font size?

**Comment by afercia** [comment]
> Not size, but the search block has also display options in the toolbar. (Something that should be revisited also, though.)

@hanneslsm good point. Would you like to create an issue for the Search block when you have a chance? I'd suggest the design team to audit all 'display' settings for all blocks though, as I think there's more cases where the placement of settings related to 'display' is largely inconsistent. @WordPress/gutenberg-design

**Comment by afercia** [comment]
Looking a bit more into this, I'd think that before making any decision on the component to use, an evaluation should be made to consider whether the Size setting should evolve to a setting controlled by the theme. To my understanding, the current values are hardcoded:

https://github.com/WordPress/gutenberg/blob/0ae80bcb8de64df1b5b71d07191802b8c51fbf74/packages/block-library/src/social-links/edit.js#L33-L38

I'd think themes should be able to provide their own size values. In that case, the values may be more than 4, which is important to determine the control ot use. For example, the Font size picker is a toggle group and changes to a CustomSelectControl when the theme provides more than 5 values.

Also, I'd think users should be allowed to set a custom size.

**Comment by t-hamano** [comment]
Rather than the theme providing presets, what if we just let the `RangeControl` apply any size we want, like for example the Site Logo block?

![image](https://github.com/user-attachments/assets/fefbffda-d2ce-4afa-aa52-23059e2807ef)

**Comment by afercia** [comment]
Using the ToggleGroupControl as done in https://github.com/WordPress/gutenberg/pull/65729 doesn't seem ideal to me. The ToggleGroupControl is designed for very short labels like the t-shirt sizes. It really doesn't work well with longer labels. 
Also, I'm not sure I understand why the social icons should only have 4 sizes.

Screenshot in Spanish, Portuguese, Dutch:

![Screenshot 2024-10-01 at 09 15 48](https://github.com/user-attachments/assets/ddccd6f4-09e9-4bcc-a8b7-28725ba1cace)

The `RangeControl` is meant for values _in a range_. Potentially, users may want to use a value that is out of the provided  range. To me, the `RangeControl` should only be used when there is a fixed, known, range. That means the usage of `RangeControl` should be audited across the whole UI including the Site Logo.

To me, this setting should allow users to set whatever size value they want. I think a UnitControl is the most appropriate one.

**Comment by t-hamano** [comment]
Allowing custom values ​​is a bit complicated because we need to consider backward compatibility.

Can't we address this in a follow-up after moving the settings to the Inspector? However, if there is no other suitable component than the `UnitControl` component, we will need to address it at the same time.

**Comment by afercia** [comment]
> Can't we address this in a follow-up after moving the settings to the Inspector?

No objections from me but the `ToggleGroupControl` doesn't look right to me.

**Comment by t-hamano** [comment]
Another approach is to use t-shirt sizes in a `ToggleGroupControl` component. This means that the labels change but the internal values ​​remain the same. However, users may misunderstand that the values ​​defined in the font size preset also apply to the social icons.

![image](https://github.com/user-attachments/assets/76b62936-575f-416a-aae9-a7e51ad534d8)

**Comment by afercia** [comment]
The t-shirt sizes pattern is less than ideal for accessibility as the visible lable mismatch the actual accessible name. It's just a wrong pattern that hasn't been designed with accessibility in mind and should be entirely retired in the future.
There are also other technical issues with that pattern. For example, themes can provide their own custom font sizes and _names. A font size with name 'Tiny' would anyway get one of the t-shirt sizes labels e.g:
- Viusal label: `S`.
- Actual accessible name and tooltip: `Tiny`.

It's just that the t-shirt sizes are assigned in a fixed order and can't be changed, regardless of the actual size names, which is a less-than-ideal implementation.

**Comment by t-hamano** [comment]
I am not talking about font size UI. I mean building our own `ToggleGroupControl` in the Social Link block. In the UI I proposed, I do not assume that tooltips will be displayed, and there will be no mismatch between the label and the tooltip. What I am proposing is simply to use shorter labels so that the `ToggleGroupControl` does not cause layout misalignment.

**Comment by afercia** [comment]
> I do not assume that tooltips will be displayed, and there will be no mismatch between the label and the tooltip

@t-hamano thanks for clarifying. So basically you are suggesting to use the t-shirt sizes names without a tooltip. That would introduce an inconsistency in the usage of the `ToggleGroupControl` that is hard to justify and I'm not sure it woul dbe a good useer experience.

![Screenshot 2024-10-02 at 08 46 09](https://github.com/user-attachments/assets/7298a911-25eb-431b-9744-2de65eb4c5b5)

That way, the Font Size picker and the Social Icons size setting would basically look the same. However, despite their similar visuals, the tooltip behavior and more importantly the accessible name would be inconsisteng.

Visually, both will whos the t-shirt sizes abbreviations: S, M, L, etc.

However, the accessible names would be different:
Font Size Picker: Small, Medium, Large, Extra Large, Extra Extra Large 
Social Icons size: S, M, L, XL 

While the _visual_ issue would be solved, we always need to think also at semantics and labeling. In this case, the inconsistent labeling wouldn't be ok.

Lastly, a `ToggleGroupControl` would also need a way to 'reset' to the defautl value, much like the font size picker.

**Comment by afercia** [comment]
On a related note: why in other cases that are conceptually similar the editor is using different controls? For example, the Image block 'resolution' is basically what users will associate to the concept of 'size', but it uses a select:

![Screenshot 2024-10-02 at 09 11 22](https://github.com/user-attachments/assets/d26c7843-6270-4171-87b3-2ab30a41dce6)

I think that for consistency and a more cohesive user experience, all settings related to the concept of 'size' should use the same UI. I'll create a new issue.

**Comment by t-hamano** [comment]
> a `ToggleGroupControl` would also need a way to 'reset' to the defautl value, much like the font size picker.

The `ToggleGroupControl` component itself has an option to deselect it ([`isDeselectable` prop](https://github.com/WordPress/gutenberg/blob/3c77f80cd3492eedc656e12d341ed2fe64ab8514/packages/components/src/toggle-group-control/toggle-group-control/README.md#isdeselectable-boolean)).

However, the text options + `isDeselectable` pattern does not seem to be recommended (See [this comment](https://github.com/WordPress/gutenberg/pull/65346#discussion_r1762374255)).

Furthermore, I don't think a reset or deselect option is necessary for now, since the current Size option don't allow deselection.

This is one of the reasons I suggested `SelectControl` component.

**Comment by jameskoster** [comment]
> The RangeControl is meant for values in a range. Potentially, users may want to use a value that is out of the provided range. To me, the RangeControl should only be used when there is a fixed, known, range

This sounds identical to `SpacingSizesControl`, this one:

<img src="https://github.com/user-attachments/assets/ecff7091-ee7f-41d7-a28a-cfe5c1a2e22a" width="300">

It's a range with (theme-supplied) defined steps. But users can elect to use a totally custom value. Would that pattern work here?

**Comment by afercia** [comment]
> This sounds identical to `SpacingSizesControl`

That's basically a `SpacingInputControl` that renders with differently based on some conditions. What I don't like much of the `SpacingInputControl` is that is a block-editor ad-hoc implementation that was designed for _spacing_ settings. I'd think there's the need for a more generic component for the concept of sizes.

Also, I'd like to mention that there is some inconsistency in the 'size' settings across the editor. From an user persepctive, the image resolution is an image 'size'. However, the setting uses a select:

![Image](https://github.com/user-attachments/assets/10646acf-a544-4e31-aad0-7d9636b478c2)

I would like the editor to always provide the same control for a setting conceptually related to 'size'.

**Comment by jameskoster** [comment]
> I'd think there's the need for a more generic component for the concept of sizes.

That's what I was thinking; a new low-level component that accepts a preset range of values and displays them as a stepped range UI accordingly. It would include an option to allow users to set a custom value. `SpacingSizesControl` would be a consumer of this component, and it could be used in the Social Icons block inspector too.

> I would like the editor to always provide the same control for a setting conceptually related to 'size'.

Generally I agree but there can be nuance. The image size control you reference doesn't _always_ affect the rendering size, so feels a bit different to me. Additionally the ergonomics of range controls (drag support, single-click setting) can be superior to `select`s, especially when there are only a few options.

**Comment by t-hamano** [comment]
> I'd think there's the need for a more generic component for the concept of sizes.

Here are the related issues/PRs:

- https://github.com/WordPress/gutenberg/issues/40507
- https://github.com/WordPress/gutenberg/pull/62866

**Comment by afercia** [comment]
> Generally I agree but there can be nuance. The image size control you reference doesn't always affect the rendering size, so feels a bit different to me.

Yes I'd agree it's a little different, still I'd love to see the same control used for all 'size' settings.

> Additionally the ergonomics of range controls (drag support, single-click setting) can be superior to selects

Why? :) A select elemeent is native and it has all the usability and accessibility support we need. I wouldn't say the same for a range control.

> especially when there are only a few options.

Which brings us back to the fact we're uaneble to predict how many values a size control will have. If it's a few, than the toggle group works well. If it's more than a few, neither the toggle group and a range would work well.

**Comment by t-hamano** [comment]
It appears that the discussion here has grown much larger than the scope of this issue was originally proposed.

Perhaps implementing custom values ​​and consistent size controls across the project would take a long time and seems to require its own issue.

Could this issue simply focus on moving the settings to the InspectorControl?

**Comment by afercia** [comment]
> ould this issue simply focus on moving the settings to the InspectorControl?

Yes please, as long as a new issue is created to continue the discussion or, maybe better, we change the title and scope of this issue to not lose references to all the feedback and considerations provided here.

**Comment by t-hamano** [comment]
While the discussion happening here has been very useful, I personally don't prefer to change the title, purpose, or scope of the original issue. This is because it could confuse someone seeing this issue for the first time as they read the comments from the top of the page.

I would be happy to just discuss moving size setting to the inspector here, and continue the discussion of custom values ​​and new components in a separate issue.

**Comment by carolinan** [comment]
+1 for the select list.

**Comment by Infinite-Null** [comment]
If there is anything actionable here, I would love to help and contribute!

**Comment by hanneslsm** [comment]
> @hanneslsm good point. Would you like to create an issue for the Search block when you have a chance? 

Sorry, I must have overlooked that. I’ve now created the issue: https://github.com/WordPress/gutenberg/issues/70138

@Infinite-Null 
> If there is anything actionable here, I would love to help and contribute!

If I see it correctly we must decide on the correct component while the dropdown / select list is currently the best option. I think we can move forward with this and, if needed, change the component later.

**Comment by t-hamano** [comment]
Thank you everyone for your feedback.

For now, it looks like it's enough to move the size setting to the inspector panel and use a select list, the `SelectControl` component. Let's resume #65729.
